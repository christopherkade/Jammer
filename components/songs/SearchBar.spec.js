import { shallowMount } from '@vue/test-utils'
import cmp from './SearchBar.vue'

describe('[UNIT] Testing the /songs Search Bar component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(cmp, {
      propsData: {
        items: [],
        isAsync: false
      },
      data() {
        return {
          isLoading: true,
          results: [
            { name: 'mockedName1', artist: 'mockedArtist1' },
            { name: 'mockedName2', artist: 'mockedArtist2' }
          ],
          isOpen: true
        }
      }
    })
  })

  it('Snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component validation', () => {
    it('should render the main (parent) component', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validtion', () => {
    it('should render the loading message', () => {
      expect(wrapper.text()).toContain('Loading results...')
    })

    it('should display the result list in a pannel', () => {
      wrapper.setData({ isLoading: false })
      expect(wrapper.text()).toContain('mockedName1 - mockedArtist1')
    })
  })
})
