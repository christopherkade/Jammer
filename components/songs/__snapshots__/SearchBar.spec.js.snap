// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] Testing the /songs Search Bar component Snapshot testing 1`] = `
"<div class=\\"autocomplete\\"><div class=\\"field has-addons\\"><div class=\\"control search-control\\"><input placeholder=\\"Search for a song\\" type=\\"text\\" class=\\"input input-searchbar\\"> <ul class=\\"autocomplete-results\\"><li class=\\"loading\\">
          Loading results...
        </li></ul></div> <div class=\\"control\\"><a class=\\"button search-button\\">
        +
      </a></div></div></div>"
`;
